db/compaction/compaction_picker_universal.cc

```diff
@@ -811,8 +814,18 @@ Compaction* UniversalCompactionBuilder::PickCompactionToReduceSizeAmp() {
         cf_name_.c_str(), file_num_buf, start_index, " to reduce size amp.\n");
   }
 
+  // size of the base sorted run for size amp calculation
+  uint64_t base_sr_size = sorted_runs_.back().size;
+  size_t sr_end_idx = sorted_runs_.size() - 1;
+  if (ioptions_.preclude_last_level_data_seconds > 0) {
+    if (sorted_runs_.back().level == ioptions_.num_levels - 1) {
```

Comment:
How to deal with that the last level is still in L0? I don't think that would be precluded, right?

Reply:
yeah, it doesn't support. It doesn't support if number of level is 1 or even 2 (in which case penultimate level is going to be Level0 which we don't support).
I added an additional check `ioptions_.num_levels > 2`.